### PR TITLE
Introduce dev & lib categorization, add first non-os rule

### PR DESCRIPTION
**Pull request by janedbal**
- Check that all identifiers are prefixed with `shipmonk`
- Structure change:

```
  src
     Rule
     Extension
     Visitor
  tests
     Rule
     Extension
```
->
```
  src
     lib          # all previous OS stuff
        Rule
        Extension
        Visitor
     dev          # dev-only stuff to serve for internal purposes of this library
        Rule
  tests
     lib          # tests for OS stuff
        Rule
        Extension
     dev          # dev-only tests and test-related stuff
        Rule
```

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -26,16 +26,23 @@
     },
     "autoload": {
         "psr-4": {
-            "ShipMonk\\PHPStan\\": "src/"
+            "ShipMonk\\PHPStan\\": [
+                "src/lib/",
+                "src/dev/"
```
Comment (olsavmic) [inline composer.json:31]: Do we really want to load `dev` in regular autoload? I'd expect it to be relevant only for `autoload-dev`

**Comment by xificurk** [comment]
One more point to consider - what gets exported as composer package code.

I would expect `src/dev` being stripped out of final package content, i.e. it should be excluded in https://github.com/shipmonk-rnd/phpstan-rules/blob/master/.gitattributes But then the dist version dir structure will look weird with seemingly useless `lib` subdirectory.
Moving the dev code to excluded top level directory would IMO simplify the configuration.

**Comment by janedbal** [comment]
@xificurk 

- gitattributes: Agreed & updated. 
- useless folder: I'm ok with that, imo people do not really care what is inside vendor (beside some nonsense data transfer) and this is super-minor.

**Review by olsavmic — APPROVED** [review]

File: src/dev/Rule/EnforceErrorIdentifierPrefixRule.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
+<?php declare(strict_types = 1);
+
+namespace ShipMonk\PHPStan\Rule;
```
Comment (JanTvrdik) [inline src/dev/Rule/EnforceErrorIdentifierPrefixRule.php:3]: I would prefer using different namespace, e.g.

```suggestion
namespace ShipMonk\PHPStanDev\Rule;
```

**Comment by janedbal** [comment]
> Check that all identifiers are prefixed with shipmonk

This main purpose was solved [diffently](https://github.com/shipmonk-rnd/phpstan-rules/pull/190), closing.

**Pull request closed without merging**